### ISSUE TITLE
IPAM Namespaces: Alpha-completeness

### ISSUE THREAD

**Issue by jathanism** [nautobot/nautobot#3429]
## Branched from #3417 

This is the work required to make the Namespace feature ready for alpha release.

Changes should be merged into #3378.

### To Do

```[tasklist]
### Prefix
- [x] UI: Child Prefixes nav tab in Prefix detail view (PrefixDetailTable) defaults to showing "Global" namespace, not matching that of the Prefix (should be same namespace as parent, once parenting implemented)
- [x] UI: When you go to "add child prefix" from `Prefix` detail view, it should default to the same Namespace as the parent
```

```[tasklist]
### VRF
- [x] Should be able to assign devices from `VRF` edit view
- [x] Ditch `RouteDistinguisher` and fold it back into `VRF.rd`
- [x] Rework `enforce_unique` checks (currently commented out); need to be removed entirely
- [x] `VRF.add_device(device, rd)` and `VRF.remove_device(device)` model methods to discourage use of `vrf.devices.add()`
- [x] `VRF.add_prefix(prefix)` and `VRF.remove_prefix(prefix)` model methods `vrf.prefixes.add()`
```

```[tasklist]
### Device
- [x] Should be able to assign `VRF` from `Device` edit view
```

```[tasklist]
### Interface
- [x] UI: Add `vrf` relationship (the optional FK from `Interface` to `VRF`)
```

```[tasklist]
### IPAddress
- [x] UI: Display `vrf` relationship as inherited from `Prefix`
```

**Comment by glennmatthews** [comment]
### Prefix

> UI: Child Prefixes nav tab in Prefix detail view (PrefixDetailTable) defaults to showing "Global" namespace, not matching that of the Prefix (should be same namespace as parent, once parenting implemented)

Appears to be done as part of parenting work.

> UI: When you go to "add child prefix" from `Prefix` detail view, it should default to the same Namespace as the parent

Appears to be done as part of parenting work.

**Question**: Should it be possible to specify a VRF(s) when editing a Prefix? If so that isn't yet present.

**Question**: Should the Prefix detail view list associated VRF(s)? Currently it always displays VRF "Global" even if a different Prefix-to-VRF association exists.

**Comment by gsnider2195** [comment]
Looking at branch [u/glennmatthews-3429-alpha-completeness](https://github.com/nautobot/nautobot/tree/u/glennmatthews-3429-alpha-completeness) I'm seeing the prefix table overflowing from its panel in the namespace detail view.

![image](https://user-images.githubusercontent.com/75227981/229155548-48c37f0c-d34e-4922-ae0f-9f450cfea47f.png)

**Comment by glennmatthews** [comment]
> Looking at branch [u/glennmatthews-3429-alpha-completeness](https://github.com/nautobot/nautobot/tree/u/glennmatthews-3429-alpha-completeness) I'm seeing the prefix table overflowing from its panel in the namespace detail view.

Yeah I noticed this too. Not sure to what extent it's fixable as a CSS issue versus to what extent we just need to hide many/most of the columns of PrefixTable to let it fit; it's probably all moot in the reworked UI in any case.

**Comment by glennmatthews** [comment]
After #3496 and #3505, I think these are all addressed in the prototype branch.

**Comment by glennmatthews** [comment]
https://github.com/nautobot/nautobot/tree/prototype/jathanism-3337-ipam-namespace
